### ISSUE TITLE
keyboard_enter_char 'Delete' is failing on iOS 9.1 Simulators

### ISSUE THREAD

**Issue by rimarsh** [calabash/calabash-ios#913]
**uia('uia.keyboard().elements().firstWithName('Delete').tap()')** command fails on 9.1 simulator with an error.

**Environment:**
Calabash 0.16.4
run_loop 1.5.5
Xcode 7.1

**Steps to reproduce:**
1. Build app
2. Execute: DEBUG=1 DEVICE_TARGET='iPhone 5s (9.1)' calabash-ios console
3. Execute: start_test_server_in_background
4. Open the keyboard and enter at least one charater
5. Execute: uia('uia.keyboard().elements().firstWithName('Delete').tap()')

**Expected result:**
  * Command is executed successfully and backspace key is pressed

**Actual result:**
  * Command fails with an error

```
irb(main):003:0> uia("uia.keyboard().elements().firstWithName('Delete').tap()")
2015-11-19 14:34:22 -0800 [RunLoop:debug]: 1:uia.keyboard().elements().firstWithName('Delete').tap()
2015-11-19 14:34:22 -0800 [RunLoop:debug]: Trying write of command 1:uia.keyboard().elements().firstWithName('Delete').tap() at index 1
2015-11-19 14:34:23 -0800 [RunLoop:debug]: validate index written for index 1 ok
RuntimeError: uia action failed because: Input: uia.keyboard().elements().firstWithName('Delete').tap()
. Error: VerboseError: Cannot perform action on invalid element: UIAElementNil from target.frontMostApp().keyboard().elements()["Delete"]  tap@[native code]
eval code
eval@[native code]
global code@file:///Users/rimarsh/dev/wx2-ios-client/411AD339-4820-46F3-869C-CD261A83B17F/_run_loop.js:28494:35
	from /Users/rimarsh/.rbenv/versions/1.9.3-p550/lib/ruby/gems/1.9.1/gems/calabash-cucumber-0.16.4/lib/calabash-cucumber/uia.rb:53:in `uia'
	from (irb):3
	from /Users/rimarsh/.rbenv/versions/1.9.3-p550/bin/irb:12:in `<main>'

```

**Comment by jmoody** [comment]
@rimarsh 

What does: 

```
uia("uia.keyboard().elements().firstWithName('Delete')")
```

return?  I believe you have not found the Delete key.  Also, if you are working on in a non English environment, the name of the delete key will not be: "Delete".

Have you tried:  

```
keyboard_enter_char 'Delete'
```

As an aside, you should update your ruby version.  The next release of Calabash will not support ruby 1.9.3.

**Comment by rimarsh** [comment]
Sorry, I was using keyboard_enter_char, I was showing the underlying command which fails. The command returns empty I think, I'll check it tomorrow. I was able to work around the issue by calling uia_type_string_raw('\b') instead. I'm using an English keyboard, I think. 

As to the ruby version, yep agreed, we're working on that update.

**Comment by jmoody** [comment]
Gotcha.  I will take a look.

**Comment by jmoody** [comment]
@rimarsh This is fixed in develop and will be available with 0.17.1.

If you want to try with the bleeding edge:

```
# Update your Gemfile
gem "calabash-cucumber", :github => "calabash/calabash-ios", :branch => "develop"

$ bundle update
```
